guide: reorder how-tos and copy edits; expand on adding existing dependencies to a stage (#1914)
iterative/dvc.org: content/docs/command-reference/commit.md

  reproduce the whole pipeline. If you're sure no pipeline results would change,
  use `dvc commit` to force update the `dvc.lock` or `.dvc` files and cache.

> - In cases where we have previously executed a stage and later notice that some
>   of the files/directories used by the stage as dependencies, or created as
>   outputs, are missing from `dvc.yaml`. We can
>   [add missing dependencies/outputs to an existing stage](/docs/user-guide/how-to/add-deps-or-outs-to-a-stage)
>   without having to execute it again. Use `dvc commit` to update the `dvc.lock`
>   file and save outputs to the cache.

Author: Per https://github.com/iterative/dvc.org/pull/1914#pullrequestreview-533883376:
```suggestion
- In some cases, we have previously executed a stage, and later notice that some
  of the files/directories used by the stage as dependencies or created as
  outputs are missing from `dvc.yaml`. It is possible to
  [add missing data to an existing stage](/docs/user-guide/how-to/add-deps-or-outs-to-a-stage).
  Use `dvc commit` to update the `dvc.lock` file and save outputs to the cache.
```
Please double check the text and format it if needed.

Reviewer: Done

  reproduce the whole pipeline. If you're sure no pipeline results would change,
  use `dvc commit` to force update the `dvc.lock` or `.dvc` files and cache.

- In some cases, we have previously executed a stage, and later notice that some
  of the files/directories used by the stage as dependencies or created as
  outputs are missing from `dvc.yaml`. It is possible to
  [add missing data to an existing stage](/docs/user-guide/how-to/add-deps-or-outs-to-a-stage).

>   Use `dvc commit` to update the `dvc.lock` file and save outputs to the cache.

Reviewer 2: It is actually better and good enough! 

I would may try still to change the order though. It is possible to add ... 

but it's fine as is I think.

Author: Agree on the order of the ideas here, let me think... ⌛

>   Use `dvc commit` to update the `dvc.lock` file and save outputs to the cache.

Author: The right order of ideas here would be something like:

> It is possible to [add missing data to existing stages], and `dvc commit` can then be used to save the outputs to the cache (and update `dvc.yaml`).

- In cases where we have previously executed a stage (either by writing
  `dvc.yaml` manually and using `dvc repro`, or with `dvc run`), but later
  notice that some of the existing dependencies of the stage or output

>   files/directories it creates, which are already in the <abbr>workspace</abbr>,

Author: Need to talk about files/dirs first, deps/outs later. Something like "...notice that some of the existing files/dirs used by the stage as dependencies, or created as outputs..."

Reviewer: Done

>   creates, which are already in the <abbr>workspace</abbr>, are missing from

Author: "dependencies it creates" ? Again, let's think the changes through please.
Also, no comma was needed here.

Reviewer: Pushed new update to clarify this.

Reviewer: Also should title of the doc be same as label? https://github.com/iterative/dvc.org/pull/1914#discussion_r521781518

Author: What doc? What label? Sorry, didn't get that Q.

Reviewer: I meant that title of how-to doc should be same as label "Add Tracked Data to a Stage"? Currently the title is "Add Dependencies or Outputs to a Stage".

Author: I think we discussed that before. The label is shorter so it fits in the nav. Please always check the docs you're writing on the dev environment, BTW. You will notice important details.

If you have a better suggestion for title or label, please open a separate comment on the corresponding lines of code. Resolcing this. Thanks

>   or created as outputs, which are already in the <abbr>workspace</abbr>, are

Author: > which are

"which could be" would be more precise but OK, since this whole part is in a `, which (something,` form that is equivalent to parentheses (meaning that it doesn't change the result of the sentence).

Author: However these sentences are really long and complicated. Not easy to read...

Author: (same in run)

Reviewer 2: To be honest I'm not sure that we need anything at all after the `[add missing outputs to an existing stage]` link. That's the whole point of this item, right? Everything else doesn't add much information unless I'm missing something.

Agree with @jorgeorpinel - it feels that it should be simlpler.

Reviewer: Removed these points (https://github.com/iterative/dvc.org/pull/1914#discussion_r526539027):
`either by ..... with dvc run`, `which are already in the workspace`, `deps and outs field respectively`
 Now I think it's much simpler. Also we already have a link to `how-to` which already has all the points I removed,

Author: The latest version had some errors. I left a separate comment with a specific suggestion so we can finalize this. Resolving here

Reviewer 2: This is still too complicated to my mind. Very hard to read.

`... , but later  .... user by, which ... are (more explanation here)` - let's try to avoid this.

Let's try to keep sentence structure simple. But in this case you probably don't need even that level of details (like `outs` field?)

Reviewer 2: also an outdated comment, disregard please ^^

- In cases where we have previously executed a stage (either by writing
  `dvc.yaml` manually and using `dvc repro`, or with `dvc run`), but later
  notice that some of the existing files/directories used by the stage as
  dependencies, or created as outputs, which are already in the

>   <abbr>workspace</abbr>, are missing from `dvc.yaml` (`deps` and `outs` field

Author: > existing... which are already in

Redundant. Probably just remove the first "existing". Please double check, re-read it. Thanks

Reviewer: Done.
